toolchain/check/testdata/builtin_conversions/no_prelude/convert_class_value_to_facet_value.carbon

```diff
@@ -0,0 +1,147 @@
+// Part of the Carbon Language project, under the Apache License v2.0 with LLVM
+// Exceptions. See /LICENSE for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+// AUTOUPDATE
+// TIP: To test this file alone, run:
+// TIP:   bazel test //toolchain/testing:file_test --test_arg=--file_tests=toolchain/check/testdata/builtin_conversions/no_prelude/convert_class_value_to_facet_value.carbon
+// TIP: To dump output, run:
+// TIP:   bazel run //toolchain/testing:file_test -- --dump_output --file_tests=toolchain/check/testdata/builtin_conversions/no_prelude/convert_class_value_to_facet_value.carbon
+
+interface Goat {}
+
+class Ginger {}
+impl Ginger as Goat {}
+
+fn WalkGoat[T:! Goat](g: T) {}
```

Comment:
Note to self: add a test with a generic interface, and the generic argument comes in a later parameter:
```
fn CallGenericMethod[T:! type](U:! Generic(T), t: T) {}

fn G(t: GenericParam) { CallGenericMethod(GenericParam, t); }
```
From @zygoloid here: https://github.com/carbon-language/carbon-lang/pull/4882/files#r1939902344

Reply:
Added this test.